src/app/components/admin/page-admin-years/page-admin-years-edit/page-admin-years-edit.component.ts

```diff
@@ -0,0 +1,123 @@
+import { Component, OnInit, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
+import { FormBuilder, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { Observable, Subscription } from 'rxjs';
+import { tap } from 'rxjs/operators';
+import { User } from 'src/api/backend';
+import { EditMode } from 'src/app/models/EditMode';
+import { IconService, RoutesService } from 'src/app/services';
+import { AdminYearsService } from 'src/app/services/admin/admin-years.service';
+
+@Component({
+  selector: 'ksi-page-admin-years-edit',
+  templateUrl: './page-admin-years-edit.component.html',
+  styleUrls: ['./page-admin-years-edit.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class PageAdminYearsEditComponent implements OnInit {
+
+  editMode: EditMode;
+  EditMode = {
+    New: 'New',
+    Edit: 'Edit'
+  };
+
+  yearId: number;
+
+  form = this.fb.group({
+    index: [0, [Validators.required]],
+    year: ["", [Validators.required]],
+    sealed: [false],
+    point_pad: [null, [Validators.required]],
+    active_orgs: [[]]
+  });
+
+  adminUsers$: Observable<User[]>;
+  subscriptions: Subscription[] = [];
+
+  constructor(
+    public icon: IconService,
+    public routes: RoutesService,
+    public adminYears: AdminYearsService,
+    private fb: FormBuilder,
+    private router: Router,
+    private cdr: ChangeDetectorRef,
+  ) { }
+
+
+  ngOnInit(): void {
+    this.yearId = Number.parseInt(this.router.url.split('/').pop() || '0', 10);
+    this.editMode = this.yearId == 0 ? EditMode.New : EditMode.Update;
+
+    if (this.editMode === EditMode.Update) {
+      this.adminYears.getYearById(this.yearId).subscribe({
+        next: (year) => {
+          if (year) {
+            this.form.patchValue(year);
+          } else {
+            alert('Year not found');
+            this.router.navigate([this.routes.routes.admin.years]);
+          }
+        },
+        error: (err) => {
+          alert(`Error loading year: ${err?.message || err}`);
```

Comment:
Místo `alert` prosím používej this.modal.yesNo('discussion-thread.post.delete.confirmation') (viz https://github.com/fi-ksi/web-frontend-angular/blob/main/src/app/components/shared/discussion-thread/discussion-thread-posts/discussion-thread-posts.component.ts#L150 ), je to lépe nastylované. Případně zkus vytvořit vlastní modal utilitku jen na zobrazování popupů.